webapp/src/containers/Rhs/projectDetails/index.tsx

```diff
@@ -109,21 +121,20 @@ const ProjectDetails = (projectDetails: ProjectDetails) => {
     }, []);
 
     useEffect(() => {
-        if (subscriptionData) {
+        // Update subscription list only when it does not match with the current data
+        if (subscriptionListReturnedByApi !== subscriptionList) {
             if (showAllSubscriptions) {
-                setSubscriptionList(subscriptionData);
+                setSubscriptionList(subscriptionListReturnedByApi);
             } else {
-                setSubscriptionList(getCurrentChannelSubscriptions(subscriptionData, currentChannelId));
+                setSubscriptionList(getCurrentChannelSubscriptions(subscriptionListReturnedByApi, currentChannelId));
             }
         }
-    }, [subscriptionData, showAllSubscriptions]);
+    }, [subscriptionListReturnedByApi]);
```

Comment:
Please check if this `useEffect` is running on each render. 
(It should be running because `subscriptionListReturnedByApi` is created again and again in each render cycle. In such cases, we need to memoize the value otherwise the dependency array won't have any performance improvement.)

Reply:
nice catch